gateway/ln-gateway/src/lightning/lnd.rs

```diff
@@ -710,6 +710,76 @@ impl ILnRpcClient for GatewayLndClient {
             }),
         }
     }
+
+    async fn list_active_channels(&self) -> Result<Vec<ChannelInfo>, LightningRpcError> {
+        let mut client = self.connect().await?;
+
+        match client
+            .lightning()
+            .list_channels(ListChannelsRequest {
+                active_only: true,
+                inactive_only: false,
+                public_only: false,
+                private_only: false,
+                peer: vec![],
+            })
+            .await
+        {
+            Ok(response) => Ok(response
+                .into_inner()
+                .channels
+                .into_iter()
+                .map(|channel| {
+                    let channel_size_sats = channel.capacity.try_into().expect("i64 -> u64");
+
+                    let local_balance_sats: u64 =
+                        channel.local_balance.try_into().expect("i64 -> u64");
+                    let local_channel_reserve_sats: u64 = match channel.local_constraints {
+                        Some(constraints) => constraints.chan_reserve_sat,
+                        None => 0,
+                    };
+
+                    let outbound_liquidity_sats =
+                        if local_balance_sats >= local_channel_reserve_sats {
+                            // We must only perform this subtraction if the local balance is
+                            // greater than or equal to the channel reserve, otherwise we would
+                            // underflow and panic.
+                            local_balance_sats - local_channel_reserve_sats
+                        } else {
+                            0
+                        };
```

Comment:
[nit] We can simplify with

```rust
let outbound_liquidity_sats = local_balance_sats.saturating_sub(local_channel_reserve_sats);
```

Same for `inbound_liquidity_sats`.

Also, do you happen to know if LND allows the local or remote balance to go below the reserve?